ext/test/http/curl_http_test.cc

```diff
@@ -248,16 +248,46 @@ TEST_F(BasicCurlHttpTests, CurlHttpOperations)
   std::multimap<std::string, std::string, curl::curl_ci> m1 = {
       {"name1", "value1_1"}, {"name1", "value1_2"}, {"name2", "value3"}, {"name3", "value3"}};
   curl::Headers headers = m1;
-  curl::HttpOperation http_operations1(http_client::Method::Head, "/get", handler, headers, body,
-                                       true);
+  curl::HttpOperation http_operations1(http_client::Method::Head, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, true);
   http_operations1.Send();
 
-  curl::HttpOperation http_operations2(http_client::Method::Get, "/get", handler, headers, body,
-                                       true);
+  curl::HttpOperation http_operations2(http_client::Method::Get, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, true);
   http_operations2.Send();
 
-  curl::HttpOperation http_operations3(http_client::Method::Get, "/get", handler, headers, body,
-                                       false);
+  curl::HttpOperation http_operations3(http_client::Method::Get, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, false);
   http_operations3.Send();
   delete handler;
 }
+
+TEST_F(BasicCurlHttpTests, SendGetRequestSync)
+{
+  received_requests_.clear();
+  curl::SessionManager session_manager;
+
+  auto session = session_manager.CreateSession("127.0.0.1", HTTP_PORT);
+  auto request = session->CreateRequest();
+  request->SetUri("get/");
+  http_client::SessionState session_state;
+  auto response = session->SendRequestSync(session_state);
+  EXPECT_EQ(response->GetStatusCode(), 200);
+  EXPECT_EQ(session_state, http_client::SessionState::Response);
+}
+
+TEST_F(BasicCurlHttpTests, SendGetRequestSyncTimeout)
+{
+  received_requests_.clear();
+  curl::SessionManager session_manager;
+
+  auto session =
```

Comment:
If I need to send 2 synchronous requests, do I need to create two separate sessions?

Why do I have to specify session IP, port, and Uri separately - is there a reason to not having something like:

`oReq.open("GET", "http://www.example.org/example.txt")`

Reply:
> If I need to send 2 synchronous requests from two different threads, do I need to create two separate sessions?

`
Session` is a logical entity here for maintaining session id, ssl cert ( if needed ), callback object ( for async request) etc. HTTP is stateless protocol, so it doesn't makes sense to have ongoing session, and sending HTTP Requests over that. 

>Why do I have to specify session IP, port, and Uri separately - is there a reason to not having something like:
>oReq.open("GET", "http://www.example.org/example.txt")

Yeah, its doable, and but don't think it is related to current PR.

Reply:
Ok, I have done the modified sync requests as below:

```
  HttpClient httpClient;
  auto result = httpClient.Get(url);
  if (result){
    auto response = result.GetResponse();
  } else {
    std::cout << result.GetSessionState();
  }
```